src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTreeUI.java

```diff
@@ -818,4 +825,41 @@ public int getIconHeight(SynthContext context) {
             return height;
         }
     }
+
+    private class CollapsedIconWrapper implements SynthIcon {
+        public void paintIcon(SynthContext context, Graphics g, int x,
+                              int y, int w, int h) {
+            if (context == null) {
+                context = getContext(tree);
+                SynthGraphicsUtils.paintIcon(collapsedIcon, context, g, x, y, w, h);
+            }
+            else {
+                SynthGraphicsUtils.paintIcon(collapsedIcon, context, g, x, y, w, h);
```

Comment:
Guess it's a copy of `ExpandedIconWrapper `class, so `CollapsedIconWrapper `and `ExpandedIconWrapper` can be optimised to use common method passing in the "icon" argument..

Reply:
Should I merge the two classes into one class?

Reply:
guess you could since they are all private..

Reply:
Merged `ExpandedIconWrapper` and `CollapsedIconWrapper` class to `IconWrapper ` class.